src/test/java/org/mule/service/scheduler/internal/SchedulerThreadPoolsTestCase.java

```diff
@@ -669,7 +669,7 @@ public void customSchedulerShutdownFromWithin() throws Exception {
       new PollingProber().check(new JUnitLambdaProbe(() -> {
         assertThat("Shutdown", scheduler.isShutdown(), is(true));
         assertThat("Terminated", scheduler.isTerminated(), is(true));
-        assertThat("Destroyed", customThreadGroup.get().isDestroyed(), is(true));
+        assertThat("ActiveCount", customThreadGroup.get().activeCount(), is(0));
```

Comment:
You still need to verify that the thread group is destroyed when java is not 21.